ui/src/main/webapp/src/app/analysis-context/analysis-context-form.component.ts

```diff
@@ -225,12 +230,15 @@ export class AnalysisContextFormComponent extends FormComponent
     }
 
     onSubmit() {
+        let observable = this._analysisContextService.update(this.analysisContext);
+
         // Store the Analysis Context
-        let update = this.analysisContext.id != null;
-        let service = update ? this._analysisContextService.update : this._analysisContextService.create;
-        let contextObservable: Observable<AnalysisContext> = service.call(this._analysisContextService, this.analysisContext, this.project);
+        if (this.analysisContext.id === 0 || this.analysisContext.id === null) {
```

Comment:
I don't think so. Why would it? (if you are referring to that `analysisContextService.update` call prior this if condition, it won't be actually called until `subscribe` is called. But I can put it into else branch for clarity)